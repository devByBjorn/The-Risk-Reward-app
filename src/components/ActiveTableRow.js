import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
// import { calculatePositiveR, calculateNegativeR } from '../calculations/riskRewardCalculation'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteFirebaseTrade } from '../actions/tradeActions'

const ActiveTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    entry,
    stop,
    target,
    status,
    opened,
    negativeR,
    positiveR,
    id }) => (
    status === 'active' &&
    <tr>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{entry}</td>
      <td>{stop}</td>
      <td>{target}</td>
      <td>{opened}</td>
      <td>{negativeR}</td> {/* loss -R*/}
      <td>{positiveR}</td> {/* win +R*/}
      <td>
        <div>
          <Link to={`/edit-trade/${id}`}>
            <EditIcon />
          </Link>
          <button onClick={() => {
            dispatch(deleteFirebaseTrade({ id }))
          }}
          ><TrashIcon />
          </button>
        </div>
      </td>
    </tr>
  )

export default connect()(ActiveTableRow)