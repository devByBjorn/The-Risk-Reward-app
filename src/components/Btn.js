import React from 'react'

export const Btn = (props) => (
  <button
    type="button"
    {...props}
  >
    {props.text}</button>
)

export const SubmitBtn = (props) => (
  <button>{props.text}</button>
)

export default Btn