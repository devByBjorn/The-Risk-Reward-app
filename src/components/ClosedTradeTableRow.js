import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteTrade } from '../actions/tradeActions'




const ClosedTradeTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    opened,
    closed,
    status,
    outcome,
    rewardToRisk,
    id }) => {

  const [rowColor, setRowColor] = useState('')

  const colors = {
    win: '#7FBA00',
    loss: '#F0341B',
    scratch: '#FFB900'
  }
  const { win, loss, scratch } = colors

  useEffect(() => {
    if (outcome === 'win') {
      setRowColor(win)
    } else if (outcome === 'loss') {
      setRowColor(loss)
    } else {
      setRowColor(scratch)
    }
  })

  return (
    status === 'closed' &&

    <tr style={{ backgroundColor: rowColor }}>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{rewardToRisk}</td>
      <td className="desktop">{opened}</td>
      <td className="desktop">{closed}</td>
      <td className="desktop">{outcome}</td>
      <td className="handle-trade-row">
        <div className="handle-trade">
          <Link to={`/edit-trade/${id}`}>
            <EditIcon
              className="edit-icon icon"
            />
          </Link>
          <button
            className="btn-transparent"
            onClick={() => {
              alert('Do you want to delete this trade?')
              dispatch(deleteTrade({ id }))
            }}
          ><TrashIcon
              className="trashcan-icon icon"
            />
          </button>
        </div>
      </td>
    </tr>
  )
}
export default connect()(ClosedTradeTableRow)