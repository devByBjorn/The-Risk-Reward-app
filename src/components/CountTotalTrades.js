import React from 'react'
import { connect } from 'react-redux'

const ClosedTradesTotal = ({ trades }) => (
  <h2>Total Closed Trades: {trades.length}</h2>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades.filter((trade) => trade.status === 'closed')
  }
}

export default connect(mapStateToProps)(ClosedTradesTotal)