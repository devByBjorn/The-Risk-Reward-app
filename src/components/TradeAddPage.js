import React from 'react'
import { connect } from 'react-redux'
import { addFirebaseTrade } from '../actions/tradeActions'
import FormParent from './FormParent'
//import FormParentTwo from './FormParentTwo'

export const TradeAddPage = ({ addFirebaseTrade, history }) => {
  const handleAddTrade = (trade) => {
    addFirebaseTrade(trade)
    history.push('/trades')
  }

  return (
    <div>
      <FormParent
        handleSubmit={handleAddTrade}
      />
    </div>
  )
}


const mapDispatchToProps = (dispatch) => ({
  addFirebaseTrade: (trade) => dispatch(addFirebaseTrade(trade))
})

export default connect(null, mapDispatchToProps)(TradeAddPage)