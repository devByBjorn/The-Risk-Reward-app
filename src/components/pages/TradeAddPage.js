import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addTrade } from '../../actions/tradeActions'
import MainParentForm from '../forms/MainParentForm'


class TradeAddFirstPage extends React.Component {
  handleAddTrade = (trade) => {
    this.props.addTrade(trade)
    this.props.history.push('/trades')
  }
  render() {
    return (
      <div>
        <MainParentForm
          handleSubmit={this.handleAddTrade}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  addTrade: (trade) => dispatch(addTrade(trade))
})

export default connect(null, mapDispatchToProps)(TradeAddFirstPage)