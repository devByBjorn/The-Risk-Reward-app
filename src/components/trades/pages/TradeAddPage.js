import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addTrade } from '../../../actions/tradeActions'
import TradeParentForm from '../forms/TradeParentForm'


class TradeAddFirstPage extends React.Component {
  handleAddTrade = (trade) => {
    this.props.addTrade(trade)
    this.props.history.push('/trades')
  }
  render() {
    return (
      <div>
        <TradeParentForm
          handleSubmit={this.handleAddTrade}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  addTrade: (trade) => dispatch(addTrade(trade))
})

export default connect(null, mapDispatchToProps)(TradeAddFirstPage)