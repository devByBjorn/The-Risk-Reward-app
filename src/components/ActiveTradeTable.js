import React from 'react'
import { connect } from 'react-redux'
import ActiveTableRow from './ActiveTableRow'
import { ThMobileAndDesktop, ThOnlyDesktop } from './TradeTableHead'
const ActiveTradeTable = ({ trades }) => (
  <div>
    <table className="table-grid active">
      <caption>Active Trades</caption>
      <thead>
        <tr>
          <ThMobileAndDesktop heading='Market' />
          <ThMobileAndDesktop heading='Direction' />
          <ThMobileAndDesktop heading='Setup' />
          <ThMobileAndDesktop heading='Entry' />
          <ThMobileAndDesktop heading='Stop' />
          <ThMobileAndDesktop heading='Target' />
          <ThOnlyDesktop heading='Opened' />
          <ThMobileAndDesktop heading='-R' />
          <ThMobileAndDesktop heading='+R' />
        </tr>
      </thead>
      <tbody>
        {trades.map((trade) =>
          (<ActiveTableRow
            {...trade}
            key={trade.id}
          />)
        )}
      </tbody>
    </table>
  </div>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades,
  }
}

export default connect(mapStateToProps)(ActiveTradeTable)