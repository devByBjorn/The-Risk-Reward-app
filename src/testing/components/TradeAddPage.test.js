import React from 'react'
import { shallow } from 'enzyme'
import { TradeAddPage } from '../../components/TradeAddPage'
import { trades } from '../fixtures/trades'

let addFirebaseTrade, history, wrapper

beforeEach(() => {
  addFirebaseTrade = jest.fn()
  history = { push: jest.fn() }
  wrapper = shallow(
    <TradeAddPage
      addFirebaseTrade={addFirebaseTrade}
      history={history}
      trade={trades[1]}
    />
  )
})

test('should render TradeAddPage correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should add trade to trades array', () => {
  wrapper.find('FormParent').prop('handleSubmit')(trades[1])
  expect(history.push).toHaveBeenLastCalledWith('/trades')
  expect(addFirebaseTrade).toHaveBeenLastCalledWith(trades[1])
})
