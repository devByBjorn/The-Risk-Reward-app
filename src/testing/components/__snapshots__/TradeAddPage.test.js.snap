// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TradeAddPage correctly 1`] = `
<div>
  <FormParent
    handleSubmit={[Function]}
  />
</div>
`;
